tools/topology/topology1/platform/intel/intel-generic-bt.m4

```diff
@@ -61,17 +60,17 @@ define(`data_names', DAI_DATA_NAMES(LIST(`     ', "ssp_data1", "ssp_data2", "ssp
 
 # Note WB is put as default in the list
 define(`ssp_config_list_1', LIST(`',
-	`MULTI_SSP_CONFIG(hw_config1, HW_CONFIG_ID, DSP_A, SSP_CLOCK(mclk, BT_MCLK, codec_mclk_in),'
+	`MULTI_SSP_CONFIG(hw_config1, HW_CONFIG_ID, DSP_A, SSP_CLOCK(mclk, 0, codec_mclk_in),'
 		`SSP_CLOCK(bclk, 256000, codec_master, inverted),'
 		`SSP_CLOCK(fsync, 16000, codec_master),'
 		`SSP_TDM(1, 16, 1, 1),'
 		`SSP_MULTI_CONFIG_DATA(ssp_data1, 16))',
-	`MULTI_SSP_CONFIG(hw_config2, eval(HW_CONFIG_ID + 1), DSP_A, SSP_CLOCK(mclk, BT_MCLK, codec_mclk_in),'
+	`MULTI_SSP_CONFIG(hw_config2, eval(HW_CONFIG_ID + 1), DSP_A, SSP_CLOCK(mclk, 0, codec_mclk_in),'
 		`SSP_CLOCK(bclk, 128000, codec_master, inverted),'
 		`SSP_CLOCK(fsync, 8000, codec_master),'
 		`SSP_TDM(1, 16, 1, 1),'
 		`SSP_MULTI_CONFIG_DATA(ssp_data2, 16))',
-	`MULTI_SSP_CONFIG(hw_config3, eval(HW_CONFIG_ID + 2), DSP_A, SSP_CLOCK(mclk, BT_MCLK, codec_mclk_in),'
+	`MULTI_SSP_CONFIG(hw_config3, eval(HW_CONFIG_ID + 2), DSP_A, SSP_CLOCK(mclk, 0, codec_mclk_in),'
```

Comment:
Humm, can you clarify what configurations conflict?

We've seen the same type of problems recently with @kv2019i and @CBala21, it's a generic problem that the MCLK should not be tied to specific SSPs.

I am not a big fan of this zero-rate solution.

Reply:
For this case: https://github.com/thesofproject/sof/issues/7803

For chrome books, we always choose same mclk rate for both headphone and speaker so there is no problem even most of our speaker amplifiers do not need mclk at all.

I am still struggling running bt offload on my chrome book but I believe it should work.

It we don't want to set the rate zero, we could set the mclk_id to a specific number. But it's weird that an invalid mclk id with a valid rate specified in topology.

Reply:
@brentlu This is a bit problematic as ADSP is the clock consuming only in HFP BT profile. With A2DP BT offload, ADSP needs to provide BCLK, so MCLK is needed. So I think a better solution is something like this https://github.com/thesofproject/sof/pull/7741 that ignores MCLK programming if the ADSP is the clock consumer.

Reply:
The problem is that the MCLK_IN is a valid setting since TGL. In theory the DSP can take the MCLK from the external world and generate BCLK/FSYNC from that external clock.

Reply:
@plbossart yes MCLK_IN could be one source for M/N divider but I can't find the implementation in m/n driver. Did I miss something?

Reply:
@brentlu  I stand corrected. It seems we can indeed omit MCLK output for BT offload (in all profiles). This needs to be double-checked on hardware, but would seem to be the ase. And indeed, this is not needed to generate BCLK.
So looks good for BT.